dms/controllers/portal.py

```diff
@@ -75,85 +131,64 @@ def portal_my_dms(
-                "sortby": sortby,
-                "filterby": filterby,
-                "access_token": access_token,
-            }
+        request.session["my_dms_folder_history"] = dms_directory_items.ids
+        # check_access
+        res = self._dms_check_access("dms.directory", dms_directory_id, access_token)
+        if not res:
+            if access_token:
+                return request.redirect("/")
+            else:
+                return request.redirect("/my")
+        dms_directory_sudo = res
+        # dms_files_count
+        domain = [
+            ("is_hidden", "=", False),
+            ("directory_id", "=", dms_directory_id),
+        ]
+        # search
+        if search and search_in:
+            search_domain = []
+            if search_in == "name":
+                search_domain = OR([search_domain, [("name", "ilike", search)]])
+            domain += search_domain
+        # items
+        dms_file_items = (
+            request.env["dms.file"]
+            .sudo(request.env.user.id)
+            .search(domain, order=sort_br)
         )
-        if dms_directory_id:
-            # check_access
-            res = self._dms_check_access(
-                "dms.directory", dms_directory_id, access_token
-            )
-            if not res:
-                if access_token:
-                    return request.redirect("/")
-                else:
-                    return request.redirect("/my")
-
-            dms_directory_sudo = res
-            # dms_files_count
-            domain = [
-                ("is_hidden", "=", False),
-                ("directory_id", "=", dms_directory_id),
-            ]
-            # search
-            if search and search_in:
-                search_domain = []
-                if search_in == "name":
-                    search_domain = OR([search_domain, [("name", "ilike", search)]])
-                domain += search_domain
-            # items
-            items = (
-                request.env["dms.file"]
-                .with_user(request.env.user.id)
-                .search(domain, order=sort_br)
-            )
-            request.session["my_dms_file_history"] = items.ids
-            dms_parent_categories = dms_directory_sudo.with_user(
-                request.env.user.id
-            )._get_parent_categories(access_token)
-            values.update(
-                {
-                    "dms_directory": dms_directory_sudo,
-                    "dms_files": items,
-                    "dms_parent_categories": dms_parent_categories,
-                }
-            )
+        request.session["my_dms_file_history"] = dms_file_items.ids
+        dms_parent_categories = dms_directory_sudo.sudo(
+            request.env.user.id
+        )._get_parent_categories(access_token)
+        # values
+        values = {
+            "dms_directories": dms_directory_items.sudo(),
+            "page_name": "dms_directory",
+            "default_url": "/my/dms/directories",
```

Comment:
```suggestion
            "default_url": "/my/dms",
```

Reply:
In V12 fix it isn't change.

Reply:
Yeah, I know. It's not critical and I have seen it later, but better to put a valid URL in this version.